l2geth/eth/gasprice/rollup_gasprice.go

```diff
@@ -24,28 +27,30 @@ func NewRollupOracle(dataPrice *big.Int, executionPrice *big.Int) *RollupOracle
 /// data by the sequencer.
 func (gpo *RollupOracle) SuggestDataPrice(ctx context.Context) (*big.Int, error) {
 	gpo.dataPriceLock.RLock()
-	price := gpo.dataPrice
-	gpo.dataPriceLock.RUnlock()
-	return price, nil
+	defer gpo.dataPriceLock.RUnlock()
+	return gpo.dataPrice, nil
 }
 
 func (gpo *RollupOracle) SetDataPrice(dataPrice *big.Int) {
 	gpo.dataPriceLock.Lock()
-	gpo.dataPrice = dataPrice
-	gpo.dataPriceLock.Unlock()
+	defer gpo.dataPriceLock.Unlock()
```

Comment:
I don't know if i like putting the rounding inside of the oracle but it does prevent the wrong value from ever ending up inside

Reply:
We could also update the function signature such that it can return an error and if the value is not a possible gas price then we can return an error and have no side effect